### DISCUSSION TITLE
exporting API keys

### DISCUSSION THREAD

**Discussion by wallet72** [datalust/seq-tickets#1517] [Help] [answered]
Is there any way to export the API keys from our existing SEQ server, so we can migrate them to a new one? 
I've done a bit of searching around and I found a suggestion to use Seqcli, but it requires a user, password and token - which I don't have set up on our existing Seq server (we do not have authentication enabled)

What I'm trying to do it migrate the API keys from our Windows Seq server into our new Linux/Docker Seq server that will become our production/only Seq server.

**Comment by nblumhardt** [comment, marked as answer]
Howdy! Thanks for dropping us a line.

There's no way to export API keys individually; the two best options are:

 * Restore a backup from the Windows server onto the Linux one (requires that both are running the same, pretty recent, Seq version), or
 * Use `seq to-mssql` or `seq to-postgres` on the Windows server to move all of its metadata into a separate database, shut the Windows Seq instance down, and then start up the Linux one with a connection string pointing to the same database

Do either of these seem like good options?

Coordinating via `support@` may be a little easier if you need us to help you through the process; feel free to drop us a line there, if it would help.

Best regards,
Nick

**Comment by wallet72** [comment]
Hey,

 

                Yeah I might have to do the database thing.

 

                The problem I have is that when I tried to restore a backup from the Windows server onto the Linux one nothing happened – nothing was restored and no errors were reported. 

 

                If I do seq to-postgres will it move the data or make a copy? And will I have to keep it in the database from then on or can it be extracted back to the server?

**Reply by nblumhardt** [reply]
Restoring the backup from Windows to Linux should be super straightforward; I wonder if perhaps the backup file location wasn't mapped into the container where you expected?

If you'd like to have another shot at this, let me know how you're attempting it. The easiest way to go is to put the .seqbak file in the mapped data volume under `Backups/` (next to `Stream/` etc), and the put a text file `restore.sh` into `Init/` right beside it:

```sh
seqsvr config set -k storage.secretKey -v <secret key from Windows box>
seqsvr restore --from-latest
```

If you use `seq metastore to-postgresql` (sorry I mistyped the command earlier), with the `--skip-config` option, it will essentially create a copy, and leave the original server as-is.

Once the destination server is configured to use PostgreSQL, there's no way to move the data back into local storage except by restoring from backup.

Sounds like the restore-from-backup option is the way to go, if you can get it working.

I'm in AEST, can jump on a call just about anytime in business hours if needed. (Support address will get to me, if so.)

Hope this helps,
Nick
